binding.cc

```diff
@@ -8,121 +7,118 @@ namespace {
 
 using namespace Napi;
 
+// Use this when a variable or parameter is unused in order to explicitly
+// silence a compiler warning about that.
+template <typename T> inline void USE(T&&) {}
+
 typedef std::function<bool(SecCertificateRef)> CertificatePredicate;
 
 template <typename T>
-class CFPointer
-{
-  T value;
+class CFPointer {
+  T value_;
 
-public:
-  CFPointer(T val) : value(val)
-  {
+ public:
+  CFPointer(T val) : value_(val) {
     CFTypeRef value_must_be_a_CFTypeRef = val;
+    USE(value_must_be_a_CFTypeRef);
   }
```

Comment:
I'm sorta guessing from the code and variable name that this here allows to guarantee somehow that value_ is always of this generic cftyperef type, but don't really understand how exactly this works. Is it just because the assignment happens (and hence the USE helper because the variable is not really used anywhere else)?

Reply:
Yup, exactly – the assignment makes sure that the types are compatible, just like an assignment in typescript also would. The `USE()` helper is only here to suppress the compiler warning that is currently being shown because of the unused variable :)

(This could also be a single line, e.g. `USE(static_cast<CFTypeRef>(val))`, but I think this way it’s a bit clearer what’s happening)

Reply:
Got it, thanks for the explanation! And yeah, having it that way is very clear, the fact that I was able to understand it without even knowing for sure how this code works is a good confirmation 😄